include/gridtools/storage/data_store.hpp

```diff
@@ -46,9 +46,9 @@ namespace gridtools {
                 std::string m_name;
                 storage::info<N> m_info;
                 traits::target_ptr_type<Traits, mutable_data_t> m_target_ptr_holder;
-                mutable_data_t *m_target_ptr;
```

Comment:
In the extended storage swap() we are swapping this ptr. We should check more carefully what's actually swapped.